http/oidc/src/main/java/org/wildfly/security/http/oidc/TokenValidator.java

```diff
@@ -288,6 +296,31 @@ private static String getAccessTokenHash(String accessTokenString, String jwsAlg
 
     }
 
+
+    private static class NonceValidator implements ErrorCodeValidator {
+        private OidcHttpFacade facade;
+
+        public NonceValidator(OidcHttpFacade facade) {
+            this.facade = facade;
+        }
+
+        public ErrorCodeValidator.Error validate(JwtContext jwtContext) throws MalformedClaimException {
+            JwtClaims idJwtClaims = jwtContext.getJwtClaims();
+            IDToken idToken = new IDToken(idJwtClaims);
+
+            OidcHttpFacade.Cookie cookie = this.facade.getRequest().getCookie(SESSION_RANDOM_VALUE);
+            if (cookie != null) {
```

Comment:
Since we should always be sending the nonce in the request, if the cookie is null, I think the validator should fail.

Reply:
done